### PR TITLE
WIP: Poo tracker

### PR DESCRIPTION
**Pull request by sbryant**
A stupid timer to check a redis key for a message. 

Remaining stuff: 

- [x] Checking a list
- [ ] ~~rpopping to clean up entries~~ not quite that stupid
- [ ] A little less garbage
- [x] 👀 at Haty's process env setup 

/cc @sshirokov to yell at me about how garbage this is and @juliamae from where all the poop-information comes from

### CODE REVIEW

File: scripts/poo-tracker.coffee [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  setInterval ->
+    checkRedisForShit()
+  , 1000
+
+  checkRedisForShit ->
+    redis_clent.get POO_TRACKER_KEY -> (err, reply)
+      return console.error("Failed get of #{POO_TRACKER_KEY}: #{err}") if err
+      return if reply == null
+
+      robot.messageRoom "#arrakis", "We got a poo over here: #{reply}"
```
Comment (juliamae) [inline scripts/poo-tracker.coffee]: FYI the message right now is "Hi! This is Mike! I'm currently pooping!" It's the custom message he configured. IDK if it makes sense with "We got a poo over here". Eventually I'll stick a URL on the end of whatever the custom message is.

Comment (sbryant) [inline scripts/poo-tracker.coffee]: 🆒  I was just taking 🔪 s in the dark

**Comment by sbryant** [comment]
I think this more or less does all the things now.

File: scripts/poo-tracker.coffee [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+redis_client = Redis.createClient(info.port, info.hostname)
+redis_clent.auth info.auth.split(":")[1] if info.auth
+
+module.exports = (robot) ->
+  checkRedisForShit = ->
+    redis_clent.lindex POO_TRACKER_KEY, -1, (err, reply) ->
+      return console.error("Failed lindex with key '#{POO_TRACKER_KEY}' and index -1: #{err}") if err
+      return if reply == null
+
+      robot.messageRoom "#arrakis", "#{reply}"
```
Comment (sshirokov) [inline scripts/poo-tracker.coffee]: Isn't the room in the env?

Comment (sbryant) [inline scripts/poo-tracker.coffee]: good point, totally forgot about that

File: scripts/poo-tracker.coffee [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+      # clean up
+      redis_clent.lrem POO_TRACKER_KEY, 0
```
Comment (sshirokov) [inline scripts/poo-tracker.coffee]: Don't we lose all historical data?

**Comment by sshirokov** [comment]
So this is supposed to speak every time it sees a poop, by checking every second and popping one off, declaring it, and moving on?

If so, that's close, but it would be nice if you also stuff them into the Hayt brain. 

Do we get timestamps on the poops, or are we declaring that the timestamp is when we pop a poop off redis?

**Comment by sbryant** [comment]
> So this is supposed to speak every time it sees a poop, by checking every second and popping one off, declaring it, and moving on?

Basically. It really just gets whatever the latest is, and discards anything else in the list.




> Do we get timestamps on the poops, or are we declaring that the timestamp is when we pop a poop off redis?

No idea. There are no times, just announcements

File: scripts/poo-tracker.coffee (hunk 3 above)
Comment (sbryant) [inline scripts/poo-tracker.coffee]: I mean we can copy but I'm not sure what the goal is

File: scripts/poo-tracker.coffee (hunk 2 above)
Comment (sbryant) [inline scripts/poo-tracker.coffee]: Should be good as of 29c4f91

File: scripts/poo-tracker.coffee [hunk 4]
```diff
@@ -0,0 +1,32 @@
+Redis = require 'redis'
+Url = require 'url'
+
+POO_TRACKER_KEY = "poops"
+POO_LATEST_KEY = "poops:latest_message"
+
+info = Url.parse process.env.POO_REDIS_URL or "redis://localhost:6379/0"
+redis_client = Redis.createClient(info.port, info.hostname)
+redis_clent.auth info.auth.split(":")[1] if info.auth
```
Comment (sshirokov) [inline scripts/poo-tracker.coffee]: Hi clent, I'm Slava.

File: scripts/poo-tracker.coffee [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+POO_TRACKER_KEY = "poops"
+POO_LATEST_KEY = "poops:latest_message"
+
+info = Url.parse process.env.POO_REDIS_URL or "redis://localhost:6379/0"
+redis_client = Redis.createClient(info.port, info.hostname)
+redis_client.auth info.auth.split(":")[1] if info.auth
+
+module.exports = (robot) ->
+  checkRedisForShit = ->
+    redis_clent.lindex POO_TRACKER_KEY, -1, (err, reply) ->
```
Comment (sshirokov) [inline scripts/poo-tracker.coffee]: Hi clent, still Slava.

File: scripts/poo-tracker.coffee [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+redis_client.auth info.auth.split(":")[1] if info.auth
+
+module.exports = (robot) ->
+  checkRedisForShit = ->
+    redis_client.lindex POO_TRACKER_KEY, -1, (err, reply) ->
+      return console.error("Failed lindex with key '#{POO_TRACKER_KEY}' and index -1: #{err}") if err
+      return if reply == null
+
+      room = process.env.HUBOT_IRC_ROOMS or "#arrakis"
+      robot.messageRoom room, "#{reply}"
```
Comment (sshirokov) [inline scripts/poo-tracker.coffee]: Results in 
```
TypeError: Cannot call method 'match' of undefined
  at Client._splitLongLines (/home/cgp/arrakis-hubot/node_modules/hubot-irc/node_modules/irc/lib/irc.js:929:11)
```

Comment (sbryant) [inline scripts/poo-tracker.coffee]: Changed this to robot.say, which according to `hubot-irc` takes a string and not a hash. So maybe give his another shot?

File: scripts/poo-tracker.coffee [hunk 7]
```diff
@@ -0,0 +1,36 @@
+Redis = require 'redis'
+Url = require 'url'
+
+info = Url.parse process.env.POO_REDIS_URL or "redis://localhost:6379/0"
+redis_client = Redis.createClient(info.port, info.hostname)
+redis_client.auth info.auth.split(":")[1] if info.auth
+
+class Envelope
```
Comment (sshirokov) [inline scripts/poo-tracker.coffee]: 💀 🔫

**Comment by sshirokov** [comment]
41 commits 50 lines. 

This is the quality I desired :+1: :poop:

**Pull request merged**